### DISCUSSION TITLE
the "on_presence_update" event duplicates the response

### DISCUSSION THREAD

**Discussion by Xalwer** [DisnakeDev/disnake#1141] [Q&A]
code:
```
@commands.Cog.listener(name='on_presence_update')
async def on_presence_update(self,before,after):
    print('test')
```
the message "test" will be displayed 2 times

**Comment by shiftinv** [comment]
Presence updates are per-guild - if a user updates their status and has `n` mutual guilds with the bot, you will generally receive `n` presence updates for that user, one per guild.

**Reply by Xalwer** [reply]
oh, okay, is it possible to turn this off?

**Reply by shiftinv** [reply]
not really. The Discord gateway dispatches these events this way, you'd have to do the de-duplication yourself
